### PR TITLE
Improve documentation of sql database & elastic pool skus

### PR DESCRIPTION
**Pull request by jaredmoo**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json [hunk 1]
```diff
@@ -200,29 +200,38 @@
         },
         "x-ms-long-running-operation": true,
         "x-ms-examples": {
-          "Creates a database with minimum number of parameters.": {
-            "$ref": "./examples/CreateDatabaseMin.json"
+          "Creates a VCore database by specifying service objective name.": {
+            "$ref": "./examples/CreateVCoreDatabaseByServiceObjective.json"
           },
-          "Creates a VCore database with minimum number of parameters.": {
-            "$ref": "./examples/CreateVCoreDatabase.json"
```
Comment (nschonni) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json]: This example isn't referenced anymore. Should it be deleted?

Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json]: Nicely spotted. It was renamed, so the old file name can go away.

Comment (nschonni) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json]: I was just poking at the Avocado results 😝 The current error message isn't the most obvious though

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json [hunk 2]
```diff
@@ -200,29 +200,38 @@
...
+          "Creates a VCore database by specifying sku name and capacity.": {
+            "$ref": "./examples/CreateVCoreDatabaseBySkuNameCapacity.json"
           },
-          "Creates a database from restore with database deletion time.": {
-            "$ref": "./examples/CreateDatabaseRestoreMode.json"
+          "Creates a data warehouse by specifying DTU capacity.": {
```
Comment (saltug) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json]: We never refer to DTU in our documentation. So creating DW by DTU is not a valid use case for us.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/examples/CreateDwDatabaseByDtuCapacity.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "resourceGroupName": "Default-SQL-SouthEastAsia",
+    "serverName": "testsvr",
+    "databaseName": "testdw",
+    "api-version": "2017-03-01-preview",
+    "parameters": {
+      "location": "westus",
+      "sku": {
+        "name": "DataWarehouse",
+        "capacity": 9000
```
Comment (saltug) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/examples/CreateDwDatabaseByDtuCapacity.json]: We never refer to DTU in our documentation. So creating DW by DTU is not a valid use case for us.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json (hunk 1 above)
Comment (NelsonDaniel) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json]: Thanks for that @nschonni

**Review by NelsonDaniel — APPROVED** [review]
It will be approved as soon as the feedback is addressed. Make sure the reference to sku is correct @jaredmoo

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json [hunk 4]
```diff
@@ -801,8 +812,8 @@
       ],
       "properties": {
         "sku": {
-          "$ref": "../../../common/v1/types.json#/definitions/Sku",
-          "description": "The name and tier of the SKU."
+          "$ref": "#/definitions/Sku",
```
Comment (NelsonDaniel) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databases.json]: actually. Here is an error. The referenece is wrong. It should reference ../../../common/v1/types.json#/definitions/Sku

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]
@jaredmoo  The spec is not passing in ModelValidation. Check the log for information about that: https://dev.azure.com/azure-sdk/public/_build/results?buildId=23186

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5990 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5990/commits/15774abb97b06870a112684aea83bb316903f4f1

**Comment by jaredmoo** [comment]
Model issues are now resolved. Thx

**Comment by NelsonDaniel** [comment]
@jaredmoo looks good to me

**Review by NelsonDaniel — APPROVED** [review]

**Comment by jaredmoo** [comment]
Thanks!

**Pull request merged**